### PR TITLE
Script generator for Perturbo

### PR DESCRIPTION
**Pull request by hurricane642**
Hello everyone!
In this PR we are adding the ability to generate `Perturbo` scripts inside `Perturbopy`. For this purpose, we have created a separate module `generate_input`, which contains the necessary `yaml` files inside, as well as the code with all the functionality. So far it seems that there is no need to split it into several separate blocks.
Script generation is done with a command that can be directly called from the command line, for example:

`input_generation -c ephmat --prefix si --band_min 10`

The command becomes available immediately after installing Perturbopy.
All necessary documentation is provided for the code.

### CODE REVIEW

File: src/perturbopy/generate_input/generate_input.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        optional of mandatory parameter for the calculation
+        
+        
+
+    Returns
+    -------
+    None
+
+    """
+    module_dir = os.path.dirname(__file__)  # Папка, где находится модуль
```
Comment (ltan01) [inline src/perturbopy/generate_input/generate_input.py:239]: What does this mean?

**Comment by ltan01** [comment]
The input generation works great! I think it might be nice to change the variable INPUT_NAME to FILE_NAME? Also, in the help message, maybe we could include the default value and variable type? Maybe we could include a link to the input parameters page on the Perturbo website?

**Comment by imaliyov** [comment]
One thing is that we should change the documentation accordingly, so modify both perturbo and perturbopy websites. Ideally, we should put a short announcement on perturbo and put detailed documentation on perturbopy

https://perturbo-code.github.io/mydoc_generate_input.html

**Comment by hurricane642** [comment]
> One thing is that we should change the documentation accordingly, so modify both perturbo and perturbopy websites. Ideally, we should put a short announcement on perturbo and put detailed documentation on perturbopy
> 
> https://perturbo-code.github.io/mydoc_generate_input.html

Sure, sounds great! I'll add docs in this PR and open new one in the perturbo-code repo

**Comment by hurricane642** [comment]
> The input generation works great! I think it might be nice to change the variable INPUT_NAME to FILE_NAME? Also, in the help message, maybe we could include the default value and variable type? Maybe we could include a link to the input parameters page on the Perturbo website?

Great, sure!

**Comment by hurricane642** [comment]
Hey guys, I've added documentation and the ideas which @ltan01 mentioned. I'll work soon on the news for the perturbo-code webpage. 
One thing that left - it probably would be a good idea to have a page with the typical scripts, as it currently exists on the https://perturbo-code.github.io/mydoc_generate_input.html, but I'm not sure

**Comment by imaliyov** [comment]
One thing to remember is that the interactive workflow input files are generated with this script. Therefore, if it is removed from Perturbo, then the interactive workflow one should be moved to Perturbopy too. 
https://perturbo-code.github.io/mydoc_interactive_workflow.html
@hurricane642 could you try to run the interactive workflow one and to make sure it works correctly?

**Comment by hurricane642** [comment]
> One thing to remember is that the interactive workflow input files are generated with this script. Therefore, if it is removed from Perturbo, then the interactive workflow one should be moved to Perturbopy too. https://perturbo-code.github.io/mydoc_interactive_workflow.html @hurricane642 could you try to run the interactive workflow one and to make sure it works correctly?

It is definitely not :(. The problem is that it's generated for the different format of documentation (for perturbopy docs and perturbo docs we use different pipelines). So, if we want to move the interactive workflow to the Perturbopy, we'll need to rewrite this script. 
On the other hand, I'm not sure that we actually need to do it. It is true that the workflow itself is created with perturbopy, but in general, perturbopy is not necessary to use it. And I think it is logical to keep it on the Perturbo page, as it will be used much more often there.

**Comment by imaliyov** [comment]
> > One thing to remember is that the interactive workflow input files are generated with this script. Therefore, if it is removed from Perturbo, then the interactive workflow one should be moved to Perturbopy too. https://perturbo-code.github.io/mydoc_interactive_workflow.html @hurricane642 could you try to run the interactive workflow one and to make sure it works correctly?
> 
> It is definitely not :(. The problem is that it's generated for the different format of documentation (for perturbopy docs and perturbo docs we use different pipelines). So, if we want to move the interactive workflow to the Perturbopy, we'll need to rewrite this script. On the other hand, I'm not sure that we actually need to do it. It is true that the workflow itself is created with perturbopy, but in general, perturbopy is not necessary to use it. And I think it is logical to keep it on the Perturbo page, as it will be used much more often there.

Interactive workflow script needs the YAML file for input parameters, and the script that generates the input files. If we move one to PerturboPy we have to move another one too...

**Comment by hurricane642** [comment]
> > > One thing to remember is that the interactive workflow input files are generated with this script. Therefore, if it is removed from Perturbo, then the interactive workflow one should be moved to Perturbopy too. https://perturbo-code.github.io/mydoc_interactive_workflow.html @hurricane642 could you try to run the interactive workflow one and to make sure it works correctly?
> > 
> > 
> > It is definitely not :(. The problem is that it's generated for the different format of documentation (for perturbopy docs and perturbo docs we use different pipelines). So, if we want to move the interactive workflow to the Perturbopy, we'll need to rewrite this script. On the other hand, I'm not sure that we actually need to do it. It is true that the workflow itself is created with perturbopy, but in general, perturbopy is not necessary to use it. And I think it is logical to keep it on the Perturbo page, as it will be used much more often there.
> 
> Interactive workflow script needs the YAML file for input parameters, and the script that generates the input files. If we move one to PerturboPy we have to move another one too...

This is true, but we can generate the html itself somewhere inside perturbopy, and then transfer it to perturbo-docs. I realize it doesn't look very simple, but it's not that much different now - we generate it in perturbo and then move it to perturbo-docs.

**Comment by imaliyov** [comment]
@hurricane642 yes, this is exactly what I meant, maybe was not clear about it :) 
We have to generate HTML in perturbopy

**Comment by hurricane642** [comment]
Oh, ok! I thought you wanted to move the interactive workflow page itself inside perturbopy! Now I see your point, ok, I'll check it out!

**Comment by hurricane642** [comment]
Hey guys, I've moved all perturbo-docs scripts to the perturbopy repo. I'll need to separately make an update in the perturbo-docs and perturbo repos, but in the perturbopy, I think, we are done

File: src/perturbopy/generate_input/input_parameters_perturbo.yml
Comment (ltan01) [inline src/perturbopy/generate_input/input_parameters_perturbo.yml:1]: I noticed that this file (and input_parameters_qe2pert.yml) seems to be an older version of the file that's being removed in the Perturbo-dev repo. I'm not sure if it's updated with everyone's new variables, but can I copy over the variables from the other file?

**Review by imaliyov — APPROVED** [review]
I approve the PR. As soon as we can run 2 Python script (if I remember correctly), to get:
1) input parameters HTML docs
2) f90 autogenerated files 
3) interactive workflow

And we can updated those files in:
1) Perturbo website,
2) perturbo-dev source code

We should be all good as those are the sole purpose of the scripts :)

File: src/perturbopy/generate_input/input_parameters_perturbo.yml
Comment (hurricane642) [reply src/perturbopy/generate_input/input_parameters_perturbo.yml:1]: I think, we'll need to make the update after the release of Perturbo. Currently, I took the parameter files from the `perturbo` repo. So our dev-repo we'll be updated also after release already

**Pull request merged**